pkg/providers/vsphere/virtualmachine/backup.go

```diff
@@ -106,6 +117,58 @@ func BackupVirtualMachine(opts BackupVirtualMachineOptions) error {
 		})
 	}
 
+	if pkgcfg.FromContext(opts.VMCtx).Features.VMIncrementalRestore {
+		curBackup, _ := curExCfg.GetString(vmopv1.VMResourceYAMLExtraConfigKey)
+		vmBackupInSync, err := isVMBackupUpToDate(opts.VMCtx.VM, curBackup)
```

Comment:
Hmm.  Why not modify the `getDesiredVMResourceYAMLForBackup` method to set the Annotation if the FSS is enabled?  That way, you don't have to call `isVMBackupUpToDate` and duplicate the marshaling bit.

Reply:
Setting the annotation and the backup version ecKey need to happen together and will happen on based on a few factors including the FSS 
factors (ie) 
`if (!ok || backupVersionAnnotation == "") || len(ecToUpdate) != 0 || !vmBackupInSync {`
and the FSS enabled.

getDesiredVMResourceYAMLForBackup will be taking on too much doing all this. We can clean up the _else_ block FSS is enabled and clean this up better.

Reply:
resolving. feel free to reopen if you feel strongly about this. we can talk about it offline.